Formula/a/autobrr.rb

```diff
@@ -0,0 +1,52 @@
+class Autobrr < Formula
+  desc "Modern, easy to use download automation for torrents and usenet"
+  homepage "https://github.com/autobrr/autobrr"
+  url "https://github.com/autobrr/autobrr/archive/refs/tags/v1.36.0.tar.gz"
+  sha256 "631ee5f858c5952e1d7a8c27b13fe33aea9021ffd814bb540f0e494291168309"
+  license "GPL-2.0-or-later"
+
+  depends_on "go" => :build
+  depends_on "node" => :build
+
+  def install
+    # setting up Go environment variables
+    ENV["GO111MODULE"] = "on"
+    ENV["GOPROXY"] = "https://proxy.golang.org,direct"
+
+    # build the frontend
+    system "npx", "pnpm", "install", "--dir", "web"
+    system "npx", "pnpm", "--dir", "web", "run", "build"
+
+    # compile flags to embed version information
+    ldflags = "-s -w -X main.version=#{version} -X main.commit=HOMEBREW"
+
+    # Backend build steps for autobrr
+    system "go", "build", "-ldflags", ldflags, "-o", bin/"autobrr", "./cmd/autobrr"
+
+    # autobrrctl
+    system "go", "build", "-ldflags", ldflags, "-o", bin/"autobrrctl", "./cmd/autobrrctl"
+  end
+
+  def post_install
+    (var/"autobrr").mkpath
+  end
+
+  service do
+    run [opt_bin/"autobrr", "--config", var/"autobrr/config.toml"]
+    keep_alive true
+    log_path var/"log/autobrr.log"
+  end
+
+  def caveats
+    <<~EOS
+      autobrr will generate a default configuration file at first run located at:
+        #{var}/autobrr/config.toml
+
+      Please modify this file as necessary after running autobrr for the first time.
+    EOS
+  end
+
+  test do
+    system "#{bin}/autobrrctl", "version"
```

Comment:
maybe we can also run some server startup test as well?

Reply:
I'll talk to the others on the team about the best way to do this. I'll keep you posted 👍

Reply:
For server tests, check for example `httpd`

Reply:
added a curl for our liveness endpoint, hope that is sufficient 😄

Reply:
Version checks have historically been awful at detecting if things work.

What's wrong with the curl test?

Reply:
Just so I'm not misunderstanding anything here. You want the test to start up the server, curl it and shut it down again, basically?